### ISSUE TITLE
Cartesian Product

### ISSUE THREAD

**Issue by marcusrossberg** [OpenLogicProject/OpenLogic#136]
Cartesian Product is only properly defined for the binary case here:

https://github.com/OpenLogicProject/OpenLogic/blob/master/content/sets-functions-relations/sets/pairs-and-products.tex

and only glossed for ternary etc. in the second example after the definition. For the last problem of that section

     Show that if $X$ has $n$ !!{element}s, then $X^k$ has $n^k$ !!{element}s.

my students got confused—but not because of the lack of precise definition of X^3, X^4, ..., X^n, ..., but rather at the other end: they didn't know what to do with X^1 and X^0.

Perhaps, when #134 is addressed, we could also add X^1 = X and X^0 = \{\emptyset\} somewhere—or is that (i) too much clutter and/or (ii) making it too easy?

**Comment by rzach** [comment]
I'm not sure how confusing it'll be to tell students that a 3-tuple <x1, x2, x3> is really a pair <<x1,x2>, x3>. At this point in the development, I thought that it would be less confusing to leave the definition as is. How about this? 

```
Define $X^k$ as the set of all $k$-tuples of !!{element}s of~$X$ for
$k > 1$. For $k = 1$, let $X^1 = X$.  Show, by induction on~$k$, that
for all $k \ge 1$, if $X$ has $n$ !!{element}s, then $X^k$ has $n^k$
!!{element}s. (Hint: for the inductive step you have to show that you
can divide $X^{k+1}$ into $n$ disjoint subsets each of which
corresponds to one of the $n^k$ !!{element}s of~$X^k$.)
```
We could later give an "official" definition, maybe after we've defined bijections -- and then we can ask the same question again, but with the focus being on bijections.

Honestly I don't know what's better. Part of me wants to do everything "properly" from the start but part of me knows that if we go full abstract at the beginning students will check out.

**Comment by rzach** [comment]
OK, after thinking about it I think the proof too confusing anyway; better to make clear how the induction is supposed to go.

**Comment by rzach** [comment]
Fixed in commit c3f4af85c2f37e900a7f2452e5e1ff63f2361f75

**Comment by marcusrossberg** [comment]
Great!
